test/common/shared.spells.test.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable camelcase */
+let shared = require('../../common/script/index.js');
+
+let newUser = (addTasks = true) => {
+  let buffs;
+  let user;
+
+  buffs = {
+    per: 0,
+    int: 0,
+    con: 0,
+    str: 0,
+    stealth: 0,
+    streaks: false,
+  };
+  user = {
+    auth: {
+      timestamps: {},
+    },
+    stats: {
+      str: 1,
+      con: 1,
+      per: 1,
+      int: 1,
+      mp: 32,
+      lvl: 1,
+      class: 'rogue',
+      buffs,
+    },
+    items: {
+      lastDrop: {
+        count: 0,
+      },
+      hatchingPotions: {},
+      eggs: {},
+      food: {},
+      gear: {
+        equipped: {},
+        costume: {},
+      },
+      quests: {},
+    },
+    party: {
+      quest: {
+        progress: {
+          down: 0,
+        },
+      },
+    },
+    preferences: {},
+    dailys: [],
+    todos: [],
+    rewards: [],
+    flags: {},
+    achievements: {},
+    contributor: {
+      level: 100,
+    },
+  };
+  shared.wrap(user);
+  user.ops.reset(null, () => {});
+  if (addTasks) {
+    _.each(['habit', 'todo', 'daily'], (task) => {
+      user.ops.addTask({
+        body: {
+          type: task,
+          id: shared.uuid(),
+        },
+      });
+    });
+  }
+  return user;
+};
+
+describe('Spells', () => {
+  let user;
+
+  beforeEach(() => {
+    user = newUser();
+    shared.wrap(user);
+  });
+
+  context('Rogue Spells', () => {
+    it('Should add exp and gp to user when backstab is used', () => {
+      let previousExp = user.stats.exp;
+      let previousGp = user.stats.gp;
+
+      shared.content.spells.rogue.backStab.cast(user, user.todos[0]);
+      expect(user.stats.exp).to.greaterThan(previousExp);
+      expect(user.stats.gp).to.greaterThan(previousGp);
+    });
+
+    it('Should pass when user is leveled passed all breaks and backstab is used', () => {
```

Comment:
Well, I can rethink how I describe this test for sure. It was ambiguous to me as well. Basically, there are many return statements that prevent the code from reaching the undefined usr._temp variable. To circumvent those stops, I simply placed the user's level at 99 and called the update function. This ensure's the usr._tmp function is called and that the full function would complete with the updated PR above. So, simply, I am testing that the function can completely work.

Let me play around with this though. I was trying to stay in context of creating a function that fails without my change and passes with the change as requested above.